connman/src/storage.c

```diff
@@ -794,11 +796,71 @@ static void keyfile_cleanup(void)
 	keyfile_hash = NULL;
 }
 
+static int validate_path(const char *pathname)
+{
+	char *resolved_path;
+	char *ret;
+	int err = 0;
+
+	resolved_path = g_try_malloc0(PATH_MAX);
```

Comment:
Why not just char resolved_path[PATH_MAX]?

Reply:
Yeah, less mem fragmentation. Removed also use of that `char *ret` as it is not actually needed anymore.